gbpservice/nfp/configurator/agents/loadbalancer_v2.py

```diff
@@ -0,0 +1,979 @@
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'listener': listener,
+                    }
+        self._send_event(lb_constants.EVENT_CREATE_LISTENER, arg_dict,
+                         serialize=True,
+                         binding_key=listener['loadbalancer_id'],
+                         key=listener['id'])
+
+    def update_listener(self, context, old_listener, listener):
+        """Enqueues event for worker to process update listener request.
+
+        :param context: RPC context
+        :param old_listener: old listener resource to be updated
+        :param listener: new listener resource
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'old_listener': old_listener,
+                    'listener': listener,
+                    }
+        self._send_event(lb_constants.EVENT_UPDATE_LISTENER, arg_dict,
+                         serialize=True,
+                         binding_key=listener['loadbalancer_id'],
+                         key=listener['id'])
+
+    def delete_listener(self, context, listener):
+        """Enqueues event for worker to process delete listener request.
+
+        :param context: RPC context
+        :param listener: listener resource to be deleted
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'listener': listener,
+                    }
+        self._send_event(lb_constants.EVENT_DELETE_LISTENER, arg_dict,
+                         serialize=True,
+                         binding_key=listener['loadbalancer_id'],
+                         key=listener['id'])
+
+    def create_pool(self, context, pool):
+        """Enqueues event for worker to process create pool request.
+
+        :param context: RPC context
+        :param pool: pool resource to be created
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'pool': pool
+                    }
+        # TODO: M:N pool is not yet implemented.
+        # For Mitaka, use binding_key=pool['listeners'][0]['id']
+        # For Liberty, use binding_key=pool['listener']['id']
+        self._send_event(lb_constants.EVENT_CREATE_POOL, arg_dict,
+                         serialize=True,
+                         binding_key=pool['loadbalancer_id'],
+                         key=pool['id'])
+
+    def update_pool(self, context, old_pool, pool):
+        """Enqueues event for worker to process update pool request.
+
+        :param context: RPC context
+        :param old_pool: old pool resource to be updated
+        :param pool: new pool resource
+
+        Returns: None
+
+        """
+        arg_dict = {'context': context,
+                    'old_pool': old_pool,
+                    'pool': pool,
+                    }
+        # For Mitaka, use binding_key=pool['listeners'][0]['id']
+        # For Liberty, use binding_key=pool['listener']['id']
```

Comment:
@njagadish this is an outdated comment. The bind_key should always be the corresponding loadbalancer.